test/distributed/org/apache/cassandra/distributed/upgrade/UpgradeTestBase.java

```diff
@@ -153,20 +189,29 @@ public TestCase upgradesFrom(Semver from)
         /** performs all supported upgrade paths that exist in between from and to (inclusive) **/
         public TestCase upgrades(Semver from, Semver to)
         {
-            SUPPORTED_UPGRADE_PATHS.stream()
-                .filter(upgradePath -> (upgradePath.left.compareTo(from) >= 0 && upgradePath.right.compareTo(to) <= 0))
-                .forEachOrdered(upgradePath ->
-                {
-                    this.upgrade.add(
-                            new TestVersions(versions.getLatest(upgradePath.left), versions.getLatest(upgradePath.right)));
-                });
+            List<TestVersions> upgrades = findUpgradePaths(from, to);
+            // order by most direct to largest path
+            Collections.sort(upgrades, Comparator.comparingInt(a -> a.upgrade.size()));
+            logger.info("Adding upgrades of\n{}", upgrades.stream().map(TestVersions::toString).collect(Collectors.joining("\n")));
```

Comment:
this isn't needed... I did this as I tried to make tests dynamic rather than running everything as a single test... I kept it assuming that if an issue exists it is likely to be in the direct path rather than the longer path (unless dealing with old data problems like 2.x -> 4.x)